FilterSimilarNoteLevels/core.js

```diff
@@ -0,0 +1,301 @@
+//==============================================================================
+// Filter Similar Note Levels v1.0
+// https://github.com/Ash-86/Filter-Similar-Note-Levels
+//
+//  Copyright (C)2023 Ashraf El Droubi (Ash-86)
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//===============================================================================
+
+
+/**
+* @param up true|false(default) Counting notes up (from bottom) or down (from top)
+* @param fullScore true|false(default) Force analysing the full score whatever is the selection
+* @param strictCounting true|false(default, original mode) In strict mode, selecting notes 1,3 from 2-notes chord, only the 1st note of that chord will be selected. In non strict mode, the notes 1,2 will be selected
+* @param expandToFullScore true(default)|false 
+*/
+function makeSelection(up, forceFullScore, strictCounting, expandToFullScore) {
+
+    /// choose counting direction /////
+    var counting = (up)?"up":"down";
+    /// expand single selection to score or measure ? ///
+    var defaultExpand=((typeof expandToFullScore === "undefined")||expandToFullScore)?"score":"measure";
+    var strict=(strictCounting)?true:false;
+    var fullScore=(fullScore)?true:false;
```

Comment:
you mean     
    var fullscore=(forceFullScore)?true:false  ?

but i noted you did not use this parameter anywhere.